disnake/enums.py

```diff
@@ -828,6 +829,438 @@ class ThreadLayout(Enum):
+    resumed = "resumed"
+    """Called when the client has resumed a session.
+    Represents the :func:`on_resumed` event.
+    """
+    shard_connect = "shard_connect"
+    """Called when a shard has successfully connected to Discord.
+    Represents the :func:`on_shard_connect` event.
+    """
+    shard_disconnect = "shard_disconnect"
+    """Called when a shard has disconnected from Discord.
+    Represents the :func:`on_shard_disconnect` event.
+    """
+    shard_ready = "shard_ready"
+    """Called when a shard has become ready.
+    Represents the :func:`on_shard_ready` event.
+    """
+    shard_resumed = "shard_resumed"
+    """Called when a shard has resumed a session.
+    Represents the :func:`on_shard_resumed` event.
+    """
+    socket_event_type = "socket_event_type"
+    """Called whenever a websocket event is received from the WebSocket.
+    Represents the :func:`on_socket_event_type` event.
+    """
+    socket_raw_receive = "socket_raw_receive"
+    """Called whenever a message is completely received from the WebSocket, before it's processed and parsed.
+    Represents the :func:`on_socket_raw_receive` event.
+    """
+    socket_raw_send = "socket_raw_send"
+    """Called whenever a send operation is done on the WebSocket before the message is sent.
+    Represents the :func:`on_socket_raw_send` event.
+    """
+    command_error = "command_error"
+    """Called when an error is raised inside a command either through user input error, check failure, or an error in your own code.
+    Represents the :func:`.on_command_error` event.
+    """
+    slash_command_error = "slash_command_error"
+    """Called when an error is raised inside a slash command either through user input error, check failure, or an error in your own code.
+    Represents the :func:`.on_slash_command_error` event.
+    """
+    user_command_error = "user_command_error"
+    """Called when an error is raised inside a user command either through check failure, or an error in your own code.
+    Represents the :func:`.on_user_command_error` event.
+    """
+    message_command_error = "message_command_error"
+    """Called when an error is raised inside a message command either through check failure, or an error in your own code.
+    Represents the :func:`.on_message_command_error` event.
+    """
+    command = "command"
+    """Called when a command is found and is about to be invoked.
+    Represents the :func:`.on_command` event.
+    """
+    slash_command = "slash_command"
+    """Called when a slash command is found and is about to be invoked.
+    Represents the :func:`.on_slash_command` event.
+    """
+    user_command = "user_command"
+    """Called when a user command is found and is about to be invoked.
+    Represents the :func:`.on_user_command` event.
+    """
+    message_command = "message_command"
+    """Called when a message command is found and is about to be invoked.
+    Represents the :func:`.on_message_command` event.
+    """
```

Comment:
We're sort of mixing `ext.commands` events into the core lib here.
imo this is fine (and somewhat unavoidable), but it would be nice to at least visually separate them from the other events by listing them at the end.

List of `ext.commands` events (in no particular order):
```py
command
command_completion
command_error
message_command
message_command_completion
message_command_error
slash_command
slash_command_completion
slash_command_error
user_command
user_command_completion
user_command_error
```